### PR TITLE
[istiod] OSSM-2294 Add --logKubernetesApiRequests option to istiod 

### PR DESCRIPTION
**Pull request by bartoszmajsak**
This option logs each individual request sent to the Kubernetes API. In addition to enabling this option, you need to set the log level to INFO.

Based on #625

### CODE REVIEW

**Review by jwendell — APPROVED** [review]
We should have upstreamed this
cc @luksa

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 1]
```diff
@@ -109,6 +109,7 @@ type Options struct {
 	// MeshServiceController is a mesh-wide service Controller.
 	MeshServiceController *aggregate.Controller
 
+	ResyncPeriod time.Duration
```
Comment (luksa) [inline pilot/pkg/serviceregistry/kube/controller/controller.go]: This looks unrelated to the logKubernetesApiRequests thing.

Comment (bartoszmajsak) [reply pilot/pkg/serviceregistry/kube/controller/controller.go]: Yeah, I couldn't really find this being used anywhere. I guess we can remove it?

Comment (bartoszmajsak) [reply pilot/pkg/serviceregistry/kube/controller/controller.go]: For whatever reason, this comes from https://github.com/maistra/istio/pull/585. Removing for now.

**Review by luksa — APPROVED** [review]
